### PR TITLE
Add test cases for log pkg

### PR DESCRIPTION
**Pull request by rahulsrivastava71**
Provide a summary of the change and which issue (i.e. ticket) is fixed.
If there are any dependencies, for example on a PR in another repository, list those.

### CODE REVIEW

File: pkg/log/vzlog/vzlog_test.go [hunk 1]
```diff
@@ -264,6 +298,79 @@ func TestZap(t *testing.T) {
...
+	rl := DefaultLogger()
+	rl.Oncef("comp1", &logger, zap.S())
+	assert.NotNil(t, rl)
+}
+
+// TestSetZapLogger is testing the call for SetZapLogger and it's not null
+func TestSetZapLogger(t *testing.T) {
+	rl := DefaultLogger()
+	rl.SetZapLogger(zap.S())
+	assert.NotNil(t, rl)
```
Comment (pfmackin) [inline pkg/log/vzlog/vzlog_test.go:323]: you need to call rl.GetZapLogger and make sure it matches what was just set

File: pkg/log/vzlog/vzlog_test.go [hunk 2]
```diff
@@ -264,6 +298,79 @@ func TestZap(t *testing.T) {
...
+}
+
+// TestDebug is testing the call for all Debug , it's not null
+func TestDebug(t *testing.T) {
+	rl := DefaultLogger()
+	rl.Debug("comp1", zap.S())
```
Comment (pfmackin) [inline pkg/log/vzlog/vzlog_test.go:329]: All of the logging with zap.S() doesn't really test anything.  You need to use the fake logger to make sure what you logged worked.

File: pkg/log/vzlog/vzlog_test.go (hunk 1 above)
Comment (rahulsrivastava71) [reply pkg/log/vzlog/vzlog_test.go:323]: this test case is only checking for setzaplogger as an individual component so to check other method on basis of this method is already done in other test case

File: pkg/log/vzlog/vzlog_test.go (hunk 2 above)
Comment (rahulsrivastava71) [reply pkg/log/vzlog/vzlog_test.go:329]: i can remove that as this , but we can keep this also as this method takes multiple arguements so i kept string and one zap object

Comment (pfmackin) [reply pkg/log/vzlog/vzlog_test.go:329]: Look at TestLogFormat how we check the message in the fake logger.  This is what you need to do for all of these tests, you have to log, then make sure the expected string got logged.

Comment (rahulsrivastava71) [reply pkg/log/vzlog/vzlog_test.go:329]: TestLogFormat is testing the EnsureLogger method which is again a different component and this is unit test cases , so the above test case is again checking the single component.

**Review by pfmackin — APPROVED** [review]

**Pull request merged**